src/common/TrackedOp.h

```diff
@@ -303,25 +304,36 @@ class TrackedOp : public boost::intrusive::list_base_hook<> {
     ++nref;
   }
   void put() {
-    if (--nref == 0) {
+    if (nref.load() == 1) {
```

Comment:
There is a race between the `load()` and `--nref` in the new `else` clause. Need to recheck in a loop.